### PR TITLE
[risk=low][RW-13148] Small changes to Workspace Publish

### PR DESCRIPTION
**Pull request by jmthibault79**
Does not resolve RW-13148.  Some fixes for small problems I noticed.

Change 500s to 404s
Optional<FeaturedWorkspaceCategory> getFeaturedCategory

Tested locally by calling endpoints and interacting with workspaces in the UI

---
**PR checklist**

- [x] I have included an issue ID or "no ticket" in the PR title as outlined in [CONTRIBUTING.md](https://github.com/all-of-us/workbench/blob/main/.github/CONTRIBUTING.md).
- [x] I have included a risk tag of the form `[risk=no|low|moderate|severe]` in the PR title as outlined in [CONTRIBUTING.md](https://github.com/all-of-us/workbench/blob/main/.github/CONTRIBUTING.md).
- [x] I have manually tested this change and my testing process is described above.
- [ ] This change includes appropriate automated tests, and I have documented any behavior that cannot be tested with code.
- [x] I have added explanatory comments where the logic is not obvious.
- One or more of the following is true:
  - [ ] This change is intended to complete a JIRA story, so I have checked that all AC are met for that story.
  - [ ] This change fixes a bug, so I have ensured the steps to reproduce are in the Jira ticket or provided above.
  - [ ] This change impacts deployment safety (e.g. removing/altering APIs which are in use), so I have documented the impacts in the description.
  - [ ] This change includes a new feature flag, so I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later.
  - [ ] This change modifies the UI, so I have taken screenshots or recordings of the new behavior and notified the PO and UX designer in [Slack](https://pmi-engteam.slack.com/archives/C02MWP2RN5P).
  - [ ] This change modifies API behavior, so I have run the relevant E2E tests locally because API changes are not covered by our PR checks.
  - [ ] None of the above apply to this change.

### CODE REVIEW

File: api/src/main/java/org/pmiops/workbench/featuredworkspace/FeaturedWorkspaceServiceImpl.java [hunk 1]
```diff
@@ -26,9 +26,12 @@ public boolean isFeaturedWorkspace(DbWorkspace dbWorkspace) {
   }
 
   @Override
-  public FeaturedWorkspaceCategory getFeaturedCategory(DbWorkspace dbWorkspace) {
-    DbFeaturedWorkspace dbFeaturedWorkspace =
-        featuredWorkspaceDao.findByWorkspace(dbWorkspace).orElseThrow();
```
Comment (jmthibault79) [inline api/src/main/java/org/pmiops/workbench/featuredworkspace/FeaturedWorkspaceServiceImpl.java:31]: `orElseThrow()` throws an exception that our router doesn't know about, which results in a 500, so we should avoid it where possible.  See `WorkspaceAdminServiceImpl.java` in this PR for a different way to handle this.

File: api/src/main/java/org/pmiops/workbench/workspaceadmin/WorkspaceAdminServiceImpl.java [hunk 2]
```diff
@@ -431,7 +431,13 @@ public DbWorkspace setPublished(
   @Override
   public void publishWorkspaceViaDB(
       String workspaceNamespace, PublishWorkspaceRequest publishWorkspaceRequest) {
-    final DbWorkspace dbWorkspace = workspaceDao.getByNamespace(workspaceNamespace).orElseThrow();
+    final DbWorkspace dbWorkspace =
+        workspaceDao
+            .getByNamespace(workspaceNamespace)
+            .orElseThrow(
+                () ->
+                    new NotFoundException(
```
Comment (jmthibault79) [inline api/src/main/java/org/pmiops/workbench/workspaceadmin/WorkspaceAdminServiceImpl.java:439]: this exception subclasses `WorkbenchException`, which means it gets auto-magically translated to a 404 error code at the controller level.

File: api/src/test/java/org/pmiops/workbench/featuredworkspace/FeaturedWorkspaceTest.java [hunk 3]
```diff
@@ -1,6 +1,7 @@
 package org.pmiops.workbench.featuredworkspace;
 
 import static com.google.common.truth.Truth.assertThat;
+import static com.google.common.truth.Truth8.assertThat;
```
Comment (jmthibault79) [inline api/src/test/java/org/pmiops/workbench/featuredworkspace/FeaturedWorkspaceTest.java:4]: needed for `hasValue()`

File: api/src/test/java/org/pmiops/workbench/featuredworkspace/FeaturedWorkspaceTest.java [hunk 4]
```diff
@@ -91,6 +92,6 @@ public void testFeaturedCategory_getFeaturedCategoryForFeaturedWorkspace() {
     when(mockFeaturedWorkspaceDao.findByWorkspace(dbWorkspace))
         .thenReturn(Optional.of(dbFeaturedWorkspace));
     assertThat(featuredWorkspaceService.getFeaturedCategory(dbWorkspace))
-        .isEqualTo(FeaturedWorkspaceCategory.TUTORIAL_WORKSPACES);
+        .hasValue(FeaturedWorkspaceCategory.TUTORIAL_WORKSPACES);
```
Comment (jmthibault79) [inline api/src/test/java/org/pmiops/workbench/featuredworkspace/FeaturedWorkspaceTest.java:95]: `hasValue()` checks the vale inside an Optional.

File: api/src/main/java/org/pmiops/workbench/utils/mappers/WorkspaceMapper.java [hunk 5]
```diff
@@ -57,9 +57,7 @@ default void setFeaturedWorkspaceCategory(
       DbWorkspace dbWorkspace,
       FeaturedWorkspaceService featuredWorkspaceService) {
     workspace.setFeaturedCategory(
-        featuredWorkspaceService.isFeaturedWorkspace(dbWorkspace)
-            ? featuredWorkspaceService.getFeaturedCategory(dbWorkspace)
-            : null);
+        featuredWorkspaceService.getFeaturedCategory(dbWorkspace).orElse(null));
```
Comment (jmthibault79) [inline api/src/main/java/org/pmiops/workbench/utils/mappers/WorkspaceMapper.java:60]: This should now work whether or not a DB entry for Featured Workspace exists

File: api/src/test/java/org/pmiops/workbench/featuredworkspace/FeaturedWorkspaceTest.java (hunk 3 above)
Comment (jmthibault79) [reply api/src/test/java/org/pmiops/workbench/featuredworkspace/FeaturedWorkspaceTest.java:4]: and `isEmpty()`

File: api/src/main/java/org/pmiops/workbench/featuredworkspace/FeaturedWorkspaceService.java [hunk 6]
```diff
@@ -1,11 +1,13 @@
 package org.pmiops.workbench.featuredworkspace;
 
+import java.util.Optional;
 import org.pmiops.workbench.db.model.DbWorkspace;
 import org.pmiops.workbench.model.FeaturedWorkspaceCategory;
 
 public interface FeaturedWorkspaceService {
 
+  // TODO: not sure if we need this, if we have the other
```
Comment (jmthibault79) [inline api/src/main/java/org/pmiops/workbench/featuredworkspace/FeaturedWorkspaceService.java:9]: should we remove this?

File: api/src/test/java/org/pmiops/workbench/utils/mappers/WorkspaceMapperTest.java [hunk 7]
```diff
@@ -171,6 +174,18 @@ public void testConvertsDbToApiWorkspace() {
 
     assertThat(ws.getCreationTime()).isEqualTo(DB_CREATION_TIMESTAMP.toInstant().toEpochMilli());
     assertThat(ws.isPublished()).isEqualTo(sourceDbWorkspace.getPublished());
+    assertThat(ws.getFeaturedCategory()).isNull();
```
Comment (jmthibault79) [inline api/src/test/java/org/pmiops/workbench/utils/mappers/WorkspaceMapperTest.java:177]: this confirms that it's safe to remove the `isFeaturedWorkspace()` check from WorkspaceMapper

**Review by Qi77Qi — APPROVED** [review]
make sense...thanks!

File: api/src/test/java/org/pmiops/workbench/api/WorkspaceAdminControllerTest.java [hunk 8]
```diff
@@ -63,8 +64,6 @@ public class WorkspaceAdminControllerTest {
   private static final String BAD_EXCEPTION_REQUEST_REASON_CHAR =
       "Locking Reason text length should be at least 10 characters long and at most 4000 characters";
 
-  @MockBean private ActionAuditQueryService mockActionAuditQueryService;
```
Comment (jmthibault79) [inline api/src/test/java/org/pmiops/workbench/api/WorkspaceAdminControllerTest.java:66]: These are not used, so I moved them into the group MockBean annotation

**Review by NehaBroad — APPROVED** [review]

**Pull request merged**